askar-crypto/src/alg/ed25519.rs

```diff
@@ -135,10 +120,7 @@ impl Debug for Ed25519KeyPair {
 impl KeyGen for Ed25519KeyPair {
     fn generate(rng: impl KeyMaterial) -> Result<Self, Error> {
         let sk = ArrayKey::<U32>::generate(rng);
-        // NB: from_bytes is infallible if the slice is the right length
-        Ok(Self::from_secret_key(
-            SecretKey::from_bytes(sk.as_ref()).unwrap(),
-        ))
+        Ok(Self::from_secret_key((&sk).try_into().unwrap()))
```

Comment:
Is this still infallible? If so, might be good to leave the comment here.

Reply:
Same here, it would only fail if `ArrayKey<U32>` can't be turned into `&[u8; 32]`